core/domain/storage_model_audit_jobs_test.py

```diff
@@ -0,0 +1,102 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for Oppia storage model audit jobs."""
+import inspect
+
+from core.domain import prod_validation_jobs_one_off
+from core.platform import models
+from core.tests import test_utils
+
+
+# This list includes statistics models and deprecated StateIdMappingModel.
+# The statistics models are included here because the audit jobs for
+# statistics models are defined in core/domain/stats_jobs_one_off.py
+# These jobs should be updated and moved to
+# core/domain/prod_validation_jobs_one_off.py and the statistics model
+# class names can then be removed from this list.
+MODEL_CLASS_NAMES_TO_EXCLUDE = {
+    'StateCounterModel',
+    'AnswerSubmittedEventLogEntryModel',
+    'ExplorationActualStartEventLogEntryModel',
+    'SolutionHitEventLogEntryModel',
+    'StartExplorationEventLogEntryModel',
+    'MaybeLeaveExplorationEventLogEntryModel',
+    'CompleteExplorationEventLogEntryModel',
+    'RateExplorationEventLogEntryModel',
+    'StateHitEventLogEntryModel',
+    'StateCompleteEventLogEntryModel',
+    'LeaveForRefresherExplorationEventLogEntryModel',
+    'ExplorationStatsModel',
+    'ExplorationIssuesModel',
+    'PlaythroughModel',
+    'LearnerAnswerDetailsModel',
+    'ExplorationAnnotationsModel',
+    'StateAnswersModel',
+    'StateAnswersCalcOutputModel',
+    'StateIdMappingModel'
+}
+
+
+class StorageModelAuditJobsTest(test_utils.GenericTestBase):
+    """Tests for Oppia storage model audit jobs."""
+
+    def test_all_models_have_audit_jobs(self):
+        all_model_names = []
+
+        # As models.NAMES is an enum, it cannot be iterated. So we use the
+        # __dict__ property which can be iterated.
+        for name in models.NAMES.__dict__:
+            if '__' not in name:
+                all_model_names.append(name)
+
+        names_of_ndb_model_classes = []
+        for name in all_model_names:
+            # We skip base models since there are no specific audit jobs
+            # for base models. The audit jobs for subclasses of base models
+            # cover the test cases for base models, so extra audit jobs
+            # for base models are not required.
+            if name == 'base_model':
+                continue
+            (module, ) = models.Registry.import_models([name])
+            for member_name, member_obj in inspect.getmembers(module):
+                if inspect.isclass(member_obj):
+                    clazz = getattr(module, member_name)
+                    if clazz.__name__ in MODEL_CLASS_NAMES_TO_EXCLUDE:
+                        continue
+                    ancestor_names = [
+                        base_class.__name__ for base_class in clazz.__bases__]
+                    if (
+                            'ndb.Model' in ancestor_names or
+                            'BaseModel' in ancestor_names or
+                            'VersionedModel' in ancestor_names):
```

Comment:
Hm, wouldn't all classes end up having ndb.Model as an ancestor? If so, you could just use that single check right?

Reply:
Done in #7101